### PR TITLE
flexible config.py

### PR DESCRIPTION
**Pull request by juliambrosman**
More flexibility in config.py for designation of config file.

### CODE REVIEW

File: genologics/config.py [hunk 1]
```diff
@@ -4,28 +4,66 @@
...
+        config_file = specified_config
+    else:
+        config = ConfigParser.SafeConfigParser()
+        try:
+            conf_file = config.read([os.path.expanduser('~/.genologicsrc'), '.genologicsrc',
+                        'genologics.conf', 'genologics.cfg', '/etc/genologics.conf'])
+
+            # First config file found wins
+            config_file = conf_file[0]
+            print(config_file)
```
Comment (Galithil) [inline genologics/config.py]: What is the purpose of this print ?

**Comment by Galithil** [comment]
I understand the idea behind the change, and since it does not break backwards compatibility, I don't have any strong arguments against this PR. 
However, I would like you to update the docstring in the beginning, as I believe that  
`you now must load genologics.config and then run:` is not correct, as you don't have to to this, this is an alternate behaviour.

**Comment by juliambrosman** [comment]
Yes, I see what you mean.  Statement in the docstring was from a previous version that wasn't backwards compatible; should've cleaned it up a bit more before the pull request.  

See changes.

File: genologics/config.py [hunk 2]
```diff
@@ -4,28 +4,57 @@
...
-	MAIN_LOG = config.get('logging', 'MAIN_LOG').rstrip()
-else:
-	MAIN_LOG = None
+'''
+Usage:
+from genologics.config import BASEURI, USERNAME, PASSWORD
+
+Alternate Usage: 
+from genologics import config
+BASEURI, USERNAME, PASSWORD, VERSION, MAIN_LOG = load_config(specified_config = <path to config file>)
```
Comment (Galithil) [inline genologics/config.py]: Don't you mean 
`BASEURI, USERNAME, PASSWORD, VERSION, MAIN_LOG = config.load_config(specified_config = <path to config file>)` ?
By using `from genologics import config` , you only get `config` in the namespace, not all the declared functions.

**Comment by juliambrosman** [comment]
yes, you're right.

**Comment by Galithil** [comment]
Okay, this looks good. 

Next question is, do you want this to be integrated in the pip package ? If so, we need to update the version.

**Comment by btupper** [comment]
Hi,

So, the alternative is that it can wait until they do their own regular integration into pip?  Otherwise, you have to install using setup.py?  This might be beyond my feeble python skills.

Cheers,
Ben
> On Mar 11, 2016, at 10:12 AM, Denis Moreno <notifications@github.com> wrote:
> 
> Okay, this looks good.
> 
> Next question is, do you want this to be integrated in the pip package ? If so, we need to update the version.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/SciLifeLab/genologics/pull/146#issuecomment-195407046>.
> 

Ben Tupper
Bigelow Laboratory for Ocean Sciences
60 Bigelow Drive, P.O. Box 380
East Boothbay, Maine 04544
http://www.bigelow.org

**Comment by juliambrosman** [comment]
So yes, if it's possible to integrate it into the pip package, that'd be most useful for us.

Thanks!

**Comment by Galithil** [comment]
Sure thing. 
If in order for that to happen, I'm going to need you to update the file genologics/version.py with the next version number, which should be 0.2.7.

Once this is done, I'll merge and forward the update to pip.

**Comment by Galithil** [comment]
Perfect.

**Comment by Galithil** [comment]
https://pypi.python.org/pypi/genologics has been updated.

**Pull request merged**